src/features/users.ts

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { User } from '../types/User';
+import { getUsers } from '../api/users';
+
+export const loadUsers = createAsyncThunk('fetch/users', () => getUsers());
+
+export const { reducer, actions } = createSlice({
+  name: 'users',
+  initialState: {
+    selectedUser: null as User | null,
+    users: [] as User[],
+  },
+  reducers: {
+    setSelectedUser: (state, { payload }: PayloadAction<User | null>) => {
+      state.selectedUser = payload;
+    },
+  },
+  extraReducers(builder) {
+    builder.addCase(
+      loadUsers.fulfilled,
+      (state, { payload }: PayloadAction<User[]>) => {
+        state.users = payload;
+      },
+    );
+  },
```

Comment:
Consider adding pending and rejected cases for `loadUsers` to manage loading and error states. This will provide better feedback to the user interface during asynchronous operations.